lib/util/itinerary.js

```diff
@@ -262,29 +262,52 @@ function getSortValues (getterFn, a, b) {
   return { aVal, bVal }
 }
 
-/**
- * Gets a comparator value for a given route's type (OTP mode).
- */
+// Lookup for the sort values associated with various OTP modes.
+// Note: JSDoc format not used to avoid bug in documentationjs.
+// https://github.com/documentationjs/documentation/issues/372
+const modeComparatorValue = {
+  SUBWAY: 1,
+  TRAM: 2,
+  RAIL: 3,
+  GONDOLA: 4,
+  FERRY: 5,
+  CABLE_CAR: 6,
+  FUNICULAR: 7,
+  BUS: 8
+}
+
+// Lookup that maps route types to the OTP mode sort values.
+// Note: JSDoc format not used to avoid bug in documentationjs.
+// https://github.com/documentationjs/documentation/issues/372
+const routeTypeComparatorValue = {
+  0: modeComparatorValue.TRAM, // - Tram, Streetcar, Light rail.
+  1: modeComparatorValue.SUBWAY, // - Subway, Metro.
+  2: modeComparatorValue.RAIL, // - Rail. Used for intercity or long-distance travel.
+  3: modeComparatorValue.BUS, // - Bus.
+  4: modeComparatorValue.FERRY, // - Ferry.
+  5: modeComparatorValue.CABLE_CAR, // - Cable tram.
+  6: modeComparatorValue.GONDOLA, // - Gondola, etc.
+  7: modeComparatorValue.FUNICULAR, // - Funicular.
+  // TODO: 11 and 12 are not a part of OTP as of 2019-02-14, but for now just
+  // associate them with bus/rail.
+  11: modeComparatorValue.BUS, // - Trolleybus.
+  12: modeComparatorValue.RAIL // - Monorail.
+}
+
+// Gets a comparator value for a given route's type (OTP mode).
+// Note: JSDoc format not used to avoid bug in documentationjs.
+// ttps://github.com/documentationjs/documentation/issues/372
 function getRouteTypeComparatorValue (route) {
-  switch (route.mode) {
-    case 'SUBWAY':
-      return 1
-    case 'TRAM':
-      return 2
-    case 'RAIL':
-      return 3
-    case 'GONDOLA':
-      return 4
-    case 'FERRY':
-      return 5
-    case 'CABLE_CAR':
-      return 6
-    case 'FUNICULAR':
-      return 7
-    case 'BUS':
-      return 8
-    default:
-      return 9999
+  // For some strange reason, the short route response in OTP returns the
+  // string-based modes, but the long route response returns the
+  // integer route type. This attempts to account for both of those cases.
+  if (typeof route.mode !== 'undefined') {
+    return modeComparatorValue[route.mode]
```

Comment:
Warning, this block can return `undefined` can can break other pieces of code that rely on = the `9999` return value...